### PR TITLE
Add  TVL

### PR DESCRIPTION
**Pull request by NeoDashboard**
**NOTE**

1. The protocol is usually listed within 24 hours of merging the PR
2. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
3. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
4. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
5. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
NUDES

##### Twitter Link:
https://twitter.com/SendnudesToken

##### Website Link:
https://nudes.army/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<circle cx="256" cy="256" r="256" fill="url(#paint0_linear_1_32)"/>
<path d="M388 300.343V240V240C388 167.098 328.902 108 256 108V108V108C183.098 108 124 167.098 124 240V240L124 300.343M388 372H369.143C337.899 372 312.571 346.672 312.571 315.429L312.572 239.999C312.571 208.756 287.244 183.429 256.001 183.429C224.757 183.429 199.429 208.757 199.429 240.001L199.429 315.429C199.429 346.672 174.101 372 142.857 372H124" stroke="white" stroke-width="37.6471" stroke-linecap="round"/>
<defs>
<linearGradient id="paint0_linear_1_32" x1="30.1176" y1="56.4706" x2="470.588" y2="451.765" gradientUnits="userSpaceOnUse">
<stop stop-color="#C63BC9"/>
<stop offset="1" stop-color="#642FFA"/>
</linearGradient>
</defs>
</svg>

##### Current TVL:
$227,862.38

##### Chain:
NeO (N3)

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
https://www.coingecko.com/en/coins/nudes

##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
https://coinmarketcap.com/currencies/nudes/

##### Short Description (to be shown on DefiLlama):
There are 7.9 billion people on Earth. Every single one of us has a duty to send NUDES to one another. The NUDES token was minted by smart contract on the Neo N3 blockchain.

##### Token address and ticker if any:
https://explorer.onegate.space/NEP17tokeninfo/0x340720c7107ef5721e44ed2ea8e314cce5c130fa
$NUDES

##### Category (full list at https://defillama.com/categories) *Please choose only one:
[Liquid Staking](https://defillama.com/protocols/Liquid%20Staking)

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
N/A

##### forkedFrom (Does your project originate from another project):
N/A

##### methodology (what is being counted as tvl, how is tvl being calculated):
TVL retrieve from NEO N3 RPC nodes

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/nudes.js: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:223:11

**Comment by NeoDashboard** [comment]
Missing logo link: https://neo-dashboard.com/imgs/logos/NUDES.png
I'm not sure how to add qs dependency or modify code :/

**Comment by llamatester** [comment]
The adapter at projects/nudes.js exports TVL: 
        
 
 
    neo                       0
    
    total                    0

**Comment by NeoDashboard** [comment]
Not sure why `node test.js projects` show total 0 I added logs localy and It seems good TVL is retrieve :)

**Comment by llamatester** [comment]
The adapter at projects/nudes.js exports TVL: 
        
 
 
    neo                       0
    
    total                    0

File: projects/nudes.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  const options = {
+      method: "post",
+      url: "http://seed2.neo.org:10332",
+      data: query,
+    };
+
+    const response = await axios(options);
+    const nudes_amount = response.data.result.stack[0].value;
+    return parseFloat(nudes_amount / 10 ** 8);
```
Comment (0xngmi) [inline projects/nudes.js:24]: this needs to return token too like 

return {nudes: number}

**Review by 0xngmi — CHANGES_REQUESTED** [review]
pls remove changes to package files

**Comment by llamatester** [comment]
The adapter at projects/nudes.js exports TVL: 
        
 
 
    neo                       0
    
    total                    0

File: projects/nudes.js (hunk 1 above)
Comment (NeoDashboard) [reply projects/nudes.js:24]: thanks :)

**Comment by llamatester** [comment]
The adapter at projects/nudes.js exports TVL: 
        
 
 
    neo                       233.35 k
    
    total                    233.35 k

File: volumes/adapters/flamingo-finance/index.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { SimpleVolumeAdapter } from "../../dexVolume.type";
+
+const { get } = require('../../../projects/helper/http')
+
+const fetch = async (timestamp: number) => {
+
+  const data = await get('https://api.flamingo.finance/project-info/defillama-volume?timestamp='+timestamp);
```
[the remark this answers is not in the document]
Comment (0xngmi) [reply volumes/adapters/flamingo-finance/index.ts:7]: changed to 24h after, @0xtawa can you confirm that this is best?

[the remark this answers is not in the document]
Comment (0xtawa) [reply volumes/adapters/flamingo-finance/index.ts:7]: Depends on what is able to return the api. The timestamp passed to the function will be last second of the day of which we want the volume, so should return 24h before. If thats not posible, what most adapters do is to call the function `getUniqStartOfTodayTimestamp` to get the 00:00:00h timestamp of that day and return the 24h after that startOfTodayTimestamp. All in UTC.

[the remark this answers is not in the document]
Comment (0xtawa) [reply volumes/adapters/flamingo-finance/index.ts:7]: @NeoDashboard as an example, if passed `1663718399` timestamp should return the volume that happened during `20/09/2022` UTC. Let me know if that's the case and I'll merge/refill and enable the adapter

[the remark this answers is not in the document]
Comment (NeoDashboard) [reply volumes/adapters/flamingo-finance/index.ts]: Okay so should be fixed now. Thanks for clarification @0xtawa

**Comment by llamatester** [comment]
The adapter at projects/nudes.js exports TVL: 
        
 
 
    neo                       233.87 k
    
    total                    233.87 k

**Comment by vrtnd** [comment]
Hi, is this tvl of staking pool? if yes then change adapter to: 
 ``` 
neo: {
    tvl: () => ({}),
    staking: your_tvl_function
  } 
```

**Pull request closed without merging**